libraries/Microsoft.Bot.Builder.Core.Extensions/FileTranscriptStore.cs

```diff
@@ -0,0 +1,216 @@
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    /// <summary>
+    /// The file transcript store stores transcripts in file system with each activity as a file.
+    /// </summary>
+    public class FileTranscriptStore : ITranscriptStore
+    {
+        private static JsonSerializerSettings jsonSettings = new JsonSerializerSettings()
+        {
+            Formatting = Formatting.Indented,
+            NullValueHandling = NullValueHandling.Ignore
+        };
+        protected string rootFolder;
+
+        public FileTranscriptStore(string folder)
+        {
+            this.rootFolder = folder ?? throw new ArgumentNullException("Missing folder");
+        }
+
+        /// <summary>
+        /// Log an activity to the transcript
+        /// </summary>
+        /// <param name="activity">activity to log</param>
+        /// <returns></returns>
+        public async Task LogActivity(IActivity activity)
+        {
+            if (activity == null)
+                throw new ArgumentNullException("activity cannot be null for LogActivity()");
+
+            var activityPath = GetActivityPath(activity);
+
+            try
+            {
+                File.WriteAllText(activityPath, JsonConvert.SerializeObject(activity, jsonSettings));
+            }
+            catch (DirectoryNotFoundException)
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(activityPath));
+                File.WriteAllText(activityPath, JsonConvert.SerializeObject(activity, jsonSettings));
+            }
+            File.SetCreationTimeUtc(activityPath, activity.Timestamp.Value.UtcDateTime);
+        }
+
+        /// <summary>
+        /// Get activity records for conversationId
+        /// </summary>
+        /// <param name="channelId"></param>
+        /// <param name="conversationId"></param>
+        /// <param name="continuationToken"></param>
+        /// <param name="startDate"></param>
+        /// <returns></returns>
+        public async Task<PagedResult<IActivity>> GetConversationActivities(string channelId, string conversationId, string continuationToken = null, DateTime startDate = default(DateTime))
+        {
+            if (channelId == null)
+                throw new ArgumentNullException($"missing {nameof(channelId)}");
+
+            if (conversationId == null)
+                throw new ArgumentNullException($"missing {nameof(conversationId)}");
+
+            var pagedResult = new PagedResult<IActivity>();
+            var conversationPath = GetConversationFolder(channelId, conversationId);
+            DirectoryInfo dir = new DirectoryInfo(conversationPath);
+
+            if (dir.Exists)
+            {
+                FileInfo[] files;
+                if (continuationToken != null)
+                {
+                    files = dir.EnumerateFiles("*.json")
+                        .OrderBy(fi => fi.CreationTimeUtc)
```

Comment:
Is there a more deterministic way to order this? I'm not sure how well this would scale, as I can envision MANY of these things, and both collisions and simple sorting may be tough.